modules/gltf/src/glb-arrow-loader.ts

```diff
@@ -0,0 +1,135 @@
+import {Matrix4} from '@math.gl/core';
+import {Accessor, Node, PlatformIO, Primitive, WebIO} from '@gltf-transform/core';
+import {KHRONOS_EXTENSIONS} from '@gltf-transform/extensions';
+import {unweld, uninstance, dequantize} from '@gltf-transform/functions';
+import * as arrow from 'apache-arrow';
+import type {LoaderWithParser, LoaderOptions} from '@loaders.gl/loader-utils';
+import {ArrowTable, DataType, Field, Schema, SchemaMetadata} from '@loaders.gl/schema';
+import {deserializeArrowField, deserializeArrowType} from '@loaders.gl/schema-utils';
+import {GLBLoader} from './glb-loader';
+
+/** GLB Arrow loader options */
+export type GLBArrowLoaderOptions = LoaderOptions & {
+  io?: PlatformIO;
+};
```

Comment:
If the application is running in a non-browser environment (Node, Deno) or requires other dependencies (Draco, Meshopt) then a custom I/O class should be provided. Example:

```javascript
import { NodeIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS } from '@gltf-transform/extensions';
import draco3d from 'draco3dgltf';

const io = new NodeIO()
  .registerExtensions(KHRONOS_EXTENSIONS)
  .registerDependencies({
    'draco3d.decoder': await draco3d.createDecoderModule()
  });
```

Reply:
Loaders.gl has a bunch of abstractions like ReadableFile etc that have implementations that work under Node, Browser, HTTPs etc. This seems to duplicate some of the IO class responsibilities.

I wonder if we could create a `LoadersIO` custom IO class for gltf-transform on top of the loaders.gl abstractions that glued the two libraries together, so that your gltf-transform would work with the abstractions we typically use in loaders. 

That way this gltf-transform based loader wouldn't become a one-off loader that doesn't fully work like other loaders do.